### DISCUSSION TITLE
Adding a single cross-platform xmake distribution binary

### DISCUSSION THREAD

**Discussion by waruqi** [xmake-io/xmake#5486] [Polls]
- Use cosmopolitan to build xmake binary
- Compress all lua files and embed into xmake binary.
- Add an executable to Github Releases, it will not modify the current default `binary + lua` files distribution.

**Comment by qudix** [comment]
This would be very useful for those who currently use and bundle premake with their source for easier building.

**Comment by SirLynix** [comment]
Maybe this version of xmake should use the project directory to store config/packages instead of user directory, so all of xmake is embedded

**Comment by waruqi** [comment]
Now we can use cosmocc to build xmake binary, and its basic features work fine. 

- https://github.com/xmake-io/xmake/pull/5489
- https://github.com/xmake-io/xmake/pull/5488
